optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -1088,89 +967,108 @@ def _from_pretrained(
             token=token,
             revision=revision,
         )
-
         if len(onnx_files) == 0:
             raise FileNotFoundError(f"Could not find any ONNX model file in {model_id}")
 
+        encoder_path = cls._infer_file_path(ENCODER_ONNX_FILE_PATTERN, encoder_file_name, onnx_files)
```

Comment:
question : do we have tests for this like https://github.com/huggingface/optimum-onnx/blob/main/tests/onnxruntime/test_decoder.py#L315 ?

Reply:
I just pushed a [commit](https://github.com/huggingface/optimum-onnx/pull/34/commits/6bf36f2fec7afe2bc7e88e957e6e8f85f69ade70) that unifies all three modeling files around the use of `_infer_file_path` which is now documented in `ORTModel._infer_file_path`.
The simplest use case is in modeling_ort, I also used it in modeling_decoder to redefine the priority logic which fixed this edge case https://github.com/huggingface/optimum-onnx/pull/34/commits/6bf36f2fec7afe2bc7e88e957e6e8f85f69ade70#diff-7d70dee5f773f702f3f375d67dc8625febb7754852e84f61f4e7ca872b7cc1e2L416

Reply:
looks great, thanks a lot @IlyasMoutawwakil